### ISSUE TITLE
checksum_url seemingly ignored

### ISSUE THREAD

**Issue by LightAxe** [voxpupuli/puppet-archive#210]
### Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: PE 2015.2.3
- Ruby: (bundled)
- Distribution: OEL 7.4 (RHEL 7.4)
- Module version: 1.1.1

### How to reproduce (e.g Puppet code you use)
  archive { "/tmp/pc.war":
    ensure        => present,
    extract       => false,
    extract_path  => '/tmp',
    source        => 'http://build/jboss/pc.war',
    checksum_url  => 'http://build/jboss/pc.war.md5',
    checksum_type => 'md5',
  }

### What are you seeing
If /tmp/pc.war already exists, nothing happens.  If /tmp/pc.war does not already exist, it is downloaded.

### What behaviour did you expect instead
If /tmp/pc.war already exists, compare its checksum to the checksum at checksum_url and redownload it if it does not match.  If /tmp/pc.war does not exist, download it.

Then validate the local checksum against the remote checksum.

### Output log
Let me know if and what log you'd like and I'll grab it!

### Any additional information you'd like to impart
The pc.war.md5 file is not being downloaded at all from the web server; I see no mention of it in --debug, and my web server's logs don't mention it being accessed (in either access_log or error_log).

I may be misinterpreting the use of the checksum_url parameter.  I'm very open to suggestions and discussion!  Ultimately, I need Puppet to recognize when a file on a webserver changes and then update the local copy.  I had hoped a simple checksum would solve that problem.

**Comment by LightAxe** [comment]
I did some more digging/experimenting and think the problem is around line 167 in `lib/puppet/type/archive.rb`:

```
  newparam(:checksum_url) do
    desc 'archive file checksum source (instead of specifying checksum)'
  end
  newparam(:digest_url) do
    desc 'archive file checksum source (instead of specifying checksum)
    (this parameter is for camptocamp/archive compatibility)'
    munge do |val|
      resource[:checksum_url] = val
    end
  end
```

My ruby-foo is very weak, but it looks like the checksum_url parameter is inadvertently ignored?

**Comment by nanliu** [comment]
No the type is just munging for compatibility so if someone uses camp to camp archive parameters it passes digest_url value to checksum_url. Based on #204 this is a regression introduced by recent changes so maybe a git bisect will help track it down.

**Comment by LightAxe** [comment]
I changed my parameters to use digest instead of checksum and it is working
as expected.

**Comment by nanliu** [comment]
@myMarck, check and see if #211 resolves this issue.

**Comment by igalic** [comment]
fixed in  #211
